crates/store/re_entity_db/src/entity_tree.rs

```diff
@@ -216,29 +121,16 @@ impl EntityTree {
         self.children.is_empty()
     }
 
-    pub fn num_children_and_fields(&self) -> usize {
-        self.children.len() + self.entity.components.len()
-    }
-
-    /// Number of timeless messages in this tree, or any child, recursively.
-    pub fn num_static_messages_recursive(&self) -> u64 {
-        self.subtree.time_histogram.num_static_messages()
-    }
-
-    pub fn time_histogram_for_component(
-        &self,
-        timeline: &Timeline,
-        component_name: impl Into<ComponentName>,
-    ) -> Option<&crate::TimeHistogram> {
-        self.entity
-            .components
-            .get(&component_name.into())
-            .and_then(|per_timeline| per_timeline.get(timeline))
+    /// Returns `false` if this entity has no children and no data.
+    pub fn is_empty(&self, chunk_store: &ChunkStore) -> bool {
+        self.children.is_empty()
+            && !chunk_store.entity_has_any_component_on_any_timeline(&self.path)
```

Comment:
Okay I'm starting to see where the ambiguities stem from.

What do you really want to know here: whether the entity has some component associated with it (which, once true, will always be true -- no matter of what the GC ends up removing), or whether the entity has some events available in the store (which can be true at some point but then false at a later time because GC)?

This method should make it clear which one of these is at play here.

We need to make that clear all over the place in all these store APIs and their docs, in fact.

Reply:
If I look at it from the perspective of the time panel, it should not display any entities which do not actually have any visible data (either static or temporal). The way I understand it, that's how it worked previously:

https://github.com/rerun-io/rerun/pull/6934/files#diff-34bc602a4376f90828813d30cf2e1ae5f940f9db3d3b51196c383b2af2cdca14L335

The component was removed from that entity in the entity tree if the histogram was empty, which means the time panel would no longer show it, and if it had no components and no children left, it would be removed completely.

I'll properly name all the APIs with `data` or `component` depending on what they're looking for, and make the distinction clear in their docs.